digdag-standards/src/main/java/io/digdag/standards/operator/td/IdOrNameParam.java

```diff
@@ -0,0 +1,112 @@
+package io.digdag.standards.operator.td;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import static com.google.common.base.Preconditions.checkState;
+
+public class IdOrNameParam
+{
+    private static final Pattern ID_PATTERN = Pattern.compile("0|[1-9][0-9]*");
+
+    private final String name;
+    private final long id;
+
+    private IdOrNameParam(String name, long id)
+    {
+        this.name = name;
+        this.id = id;
+    }
+
+    public boolean isName()
+    {
+        return name != null;
+    }
+
+    public boolean isId()
+    {
+        return name == null;
+    }
+
+    public String getName()
+    {
+        checkState(isName());
+        return name;
+    }
+
+    public long getId()
+    {
+        checkState(isId());
+        return id;
+    }
+
+    @JsonCreator
+    public static IdOrNameParam of(JsonNode expr)
+    {
+        if (expr.isIntegralNumber()) {
+            return new IdOrNameParam(null, expr.asLong());
+        }
+        else if (expr.isTextual()) {
+            String text = expr.asText();
+            if (ID_PATTERN.matcher(text).matches()) {
```

Comment:
`"1234"` will be handled as an connection id.
There're 3 reasons to do this:

* `${...}` syntax always returns string. If here uses `"1234"` as an name of connection, there're no ways to parameterize it as an id.
* when web console or external programs generate .dig files, they may not care about types strictly (e.g. text input box of web console is always text even though users put integer there). Using different behavior depending on type causes troubles with it.
* `ObjectMapper.readValue(it, long.class)` successfully converts `"1234"` into long.